File: merge_sort/go/mergesort_test.go
package mergesort_test

import (
	"slices"
	"testing"

	mergesort "github.com/alirostami1/dsa/merge_sort/go"
	"github.com/stretchr/testify/assert"
)

func TestMergeSort(t *testing.T) {
	arr := []int{5, 2, 4, 3, 1}

	correctlySortedArr := make([]int, len(arr))
	copy(correctlySortedArr, arr)
	slices.Sort(correctlySortedArr)

	mergesort.MergeSort(arr)

	assert.Equal(t, correctlySortedArr, arr)
}

File: merge_sort/typescript/mergesort.test.ts
import mergeSort from './mergesort';

test('sorting an array', () => {
    // const arr = [5, 2, 4, 3, 1]
    const arr = Array.from({ length: 40 }, () => Math.floor(Math.random() * 40));

    const correctlySortedArr = [...arr]
    correctlySortedArr.sort((a, b) => a - b)

    mergeSort(arr, 0, arr.length - 1)

    expect(arr).toEqual(correctlySortedArr);
});

